### PR TITLE
Kiraboyle/sc 42504/add or remove semantic versioning attribute

### PR DESCRIPTION
**Pull request by kcboyle**

### CODE REVIEW

File: pkg/kotsclient/channel.go [hunk 1]
```diff
@@ -160,3 +160,23 @@ func (c *VendorV3Client) ArchiveChannel(appID, channelID string) error {
 
 	return nil
 }
+
+func (c *VendorV3Client) UpdateSemanticVersioning(appID string, channel *channels.AppChannel, enableSemver bool) error {
+	request := types.UpdateChannelRequest{
```
Comment (sgalsaleh) [inline pkg/kotsclient/channel.go:165]: from the reference page, it seems like the app id, channel id and name are all required, but i only see the name passed here. https://replicated-vendor-api.readme.io/v3/reference/updatechannel

File: pkg/types/channel.go [hunk 2]
```diff
@@ -59,6 +59,15 @@ type CreateChannelRequest struct {
 	Name                       string `json:"name"`
 }
 
+type UpdateChannelRequest struct {
+	// Description of the channel that is to be created.
+	Name                  string `json:"name"`
+	SemverRequired        bool   `json:"semverRequired,omitempty"`
+	Description           string `json:"description,omitempty"`
+	ChannelIcon           string `json:"channelIcon,omitempty"`
+	AutoAirgapBuildsValue int    `json:"autoAirgapBuildsValue,omitempty"`
```
Comment (sgalsaleh) [inline pkg/types/channel.go:68]: IMO unused fields shouldn't added here

File: pkg/kotsclient/channel.go (hunk 1 above)
Comment (kcboyle) [reply pkg/kotsclient/channel.go:165]: ah. I got lazy and passed the whole channel instead. If it's more clear I'll pass them both separately

File: pkg/types/channel.go (hunk 2 above)
Comment (kcboyle) [reply pkg/types/channel.go]: done

**Review by sgalsaleh — DISMISSED** [review]

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**